src/shogun/features/DenseFeatures.cpp

```diff
@@ -630,194 +630,15 @@ GET_FEATURE_TYPE(F_DREAL, float64_t)
-
-	int32_t vlen;
-	bool vfree;
-	uint8_t* vec1 = get_feature_vector(vec_idx1, vlen, vfree);
-
-	ASSERT(vlen == num_features)
-	float64_t result = 0;
-
-	for (int32_t i = 0; i < num_features; i++)
-		result += vec1[i] * vec2[i];
-
-	free_feature_vector(vec1, vec_idx1, vfree);
-
-	return result;
-}
-
-template<> float64_t CDenseFeatures<int16_t>::dense_dot(
-		int32_t vec_idx1, const float64_t* vec2, int32_t vec2_len) const
-{
-	ASSERT(vec2_len == num_features)
-
-	int32_t vlen;
-	bool vfree;
-	int16_t* vec1 = get_feature_vector(vec_idx1, vlen, vfree);
-
-	ASSERT(vlen == num_features)
-	float64_t result = 0;
-
-	for (int32_t i = 0; i < num_features; i++)
-		result += vec1[i] * vec2[i];
-
-	free_feature_vector(vec1, vec_idx1, vfree);
-
-	return result;
-}
-
-template<> float64_t CDenseFeatures<uint16_t>::dense_dot(
-		int32_t vec_idx1, const float64_t* vec2, int32_t vec2_len) const
-{
-	ASSERT(vec2_len == num_features)
-
-	int32_t vlen;
-	bool vfree;
-	uint16_t* vec1 = get_feature_vector(vec_idx1, vlen, vfree);
-
-	ASSERT(vlen == num_features)
-	float64_t result = 0;
-
-	for (int32_t i = 0; i < num_features; i++)
-		result += vec1[i] * vec2[i];
-
-	free_feature_vector(vec1, vec_idx1, vfree);
-
-	return result;
-}
-
-template<> float64_t CDenseFeatures<int32_t>::dense_dot(
-		int32_t vec_idx1, const float64_t* vec2, int32_t vec2_len) const
-{
-	ASSERT(vec2_len == num_features)
-
-	int32_t vlen;
-	bool vfree;
-	int32_t* vec1 = get_feature_vector(vec_idx1, vlen, vfree);
-
-	ASSERT(vlen == num_features)
-	float64_t result = 0;
-
-	for (int32_t i = 0; i < num_features; i++)
-		result += vec1[i] * vec2[i];
-
-	free_feature_vector(vec1, vec_idx1, vfree);
-
-	return result;
-}
-
-template<> float64_t CDenseFeatures<uint32_t>::dense_dot(
-		int32_t vec_idx1, const float64_t* vec2, int32_t vec2_len) const
-{
-	ASSERT(vec2_len == num_features)
-
-	int32_t vlen;
-	bool vfree;
-	uint32_t* vec1 = get_feature_vector(vec_idx1, vlen, vfree);
-
-	ASSERT(vlen == num_features)
-	float64_t result = 0;
-
-	for (int32_t i = 0; i < num_features; i++)
-		result += vec1[i] * vec2[i];
-
-	free_feature_vector(vec1, vec_idx1, vfree);
-
-	return result;
-}
-
-template<> float64_t CDenseFeatures<int64_t>::dense_dot(
-		int32_t vec_idx1, const float64_t* vec2, int32_t vec2_len) const
-{
-	ASSERT(vec2_len == num_features)
-
-	int32_t vlen;
-	bool vfree;
-	int64_t* vec1 = get_feature_vector(vec_idx1, vlen, vfree);
-
-	ASSERT(vlen == num_features)
-	float64_t result = 0;
-
-	for (int32_t i = 0; i < num_features; i++)
-		result += vec1[i] * vec2[i];
-
-	free_feature_vector(vec1, vec_idx1, vfree);
-
-	return result;
-}
-
-template<> float64_t CDenseFeatures<uint64_t>::dense_dot(
-		int32_t vec_idx1, const float64_t* vec2, int32_t vec2_len) const
-{
-	ASSERT(vec2_len == num_features)
+    ASSERT(vec2_len == num_features)
 
 	int32_t vlen;
 	bool vfree;
-	uint64_t* vec1 = get_feature_vector(vec_idx1, vlen, vfree);
+	ST* vec1 = get_feature_vector(vec_idx1, vlen, vfree);
```

Comment:
minor: formatting whitespace issues

Reply:
i wonder why the clang-format job is not complaining....?
@geektoni